### DISCUSSION TITLE
url base for a collection. An idea ?

### DISCUSSION THREAD

**Discussion by vifeng** [thunderclient/thunder-client-support#814] [General]
Hi, thanks. wonderfull app.
I've looked into the doc but couldn't find the answer. Is there a way to preset a url and its port for a collection or a folder ? It would be awesome.
Thus, we would only type the route `/api/employees` instead of `{{host}}:{{port_client}}/api/employees` for each request.
cheers for your feedback
Vifeng

**Comment by rangav** [comment]
Hi @vifeng, happy to know you like to the extension.

We have base url option in `collection settings`, Right click of any collection and select `Settings` option

<img width="1226" alt="Screenshot 2022-09-30 at 15 38 02" src="https://user-images.githubusercontent.com/8637550/193294510-ac47f444-d2c4-438c-8c1e-6e7da560adbd.png">

**Reply by vifeng** [reply]
i don't have this feature :( my version is v1.19.3

**Reply by vifeng** [reply]
my mistake, I was looking into a folder... it would be nice to have the same for the folders adding up to the base url then. :) 
thanks for your prompt reply

**Reply by rangav** [reply]
If we add to folders, then things will get complicated, because there could be lot of nested folders, it will confuse users.

**Reply by vifeng** [reply]
I see, what is the best practice, then, when you have for a given project ? 
**Server side :** 
api/employee
api/employees/{id}
api2/lessons/
api2/lessons/professors

**Client side**
same same

I would have done a collection for the project, then folder for each api, employee, lesson. but then I don't take advantage of the base url. should I do a collection instead of folder ? thanks

**Reply by rangav** [reply]
it depends upon how many `nested folders` you have in any of the above and how many `api end points`.

If it's a small collection, then better keep all in one single collection.

you can also use env variables for repeated strings if you want, like below 

- {{baseEmp}} => `api/employee` 
- and {{baseLesson}} => `api2/lessons`

also depends upon what ever is easy to `manage  and test`.

I would start with single collection and see how it goes.

**Reply by vifeng** [reply]
ok I see thanks
